NotebookAnalyzers/utils.py

```diff
@@ -27,7 +27,51 @@ def notebookToCode(filename):
     return py_code
 
 def functionsNumber(code):
-    """The function takes a python code sintax tree and returns the number of function definitions"""
+    """The function takes a python code and returns the number of function definitions"""
     tree = ast.parse(code)
     f_num=sum(isinstance(exp, ast.FunctionDef) for exp in tree.body)
-    return f_num
\ No newline at end of file
+    return f_num
+
+def markdownNumber(notebook):
+    """The function takes the notebook dictionary and returns both the number of markdown wors and number of markdown titles"""
+    titles=0
+    markdowns=0
+    for cell in notebook["cells"]:
+        if cell["cell_type"]=='markdown':
+            rows=len(cell['source'])
+            for row in cell['source']:
+                if row[0:2]=='# ':
```

Comment:
Così facendo non consideri il caso in cui ci siano spazi bianchi prima dell'asterisco e tieni conto soltanto dei titoli Markdown di primo livello (quelli più in alto nella gerarchia, contraddistinti da un solo cancelletto `#`).

Ti suggerisco di adottare questa strategia:

```python
row.lstrip().startswith('#')
```

`lstrip` elimina eventuali spazi a inizio riga e `startswith` si assicura che la riga cominci con un `#`, ammettendo che continui con altri cancelletti.

A voler essere ancora più precisi, si dovrebbe controllare che dopo un numero di `#` che può variare da 1 a 6, vi sia un carattere di spazio e poi del testo. Ma per far questo bisognerebbe passare a usare le espressioni regolari.